nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -0,0 +1,186 @@
+define([
+    'uuid',
+    'common/appUtils',
+    'common/utils',
+    'common/runtime',
+    'common/busEventManager',
+    'base/js/namespace'
+], (
+    Uuid,
+    AppUtils,
+    Utils,
+    Runtime,
+    BusEventManager,
+    Jupyter
+) => {
+    'use strict';
+    const CELL_TYPE = 'app-bulk-import';
+
+    /**
+     * This class creates and manages the bulk import cell. This works with, and wraps around,
+     * the Jupyter Cell object.
+     */
+    class BulkImportCell {
+        /**
+         * The constructor does the work of initializing the bulk import cell module. It
+         * modifies the cell metadata, if the initialize parameter is truthy, and extends the
+         * cell object to do things we need it to, like respond to minimization requests and
+         * render its icon. This modification also includes the contents of typesToFiles if
+         * present.
+         *
+         * If initialize is falsy, no changes are made to the structure, even if typesToFiles is
+         * present and contains new data.
+         * @param {Cell} cell a Jupyter notebook cell. This should be a code cell, and will throw
+         * an Error if it is not.
+         * @param {boolean} initialize if true, this will initialize the bulk import cell
+         * structure that gets serialized in the cell metadata. This should ONLY be set true when
+         * creating a new bulk import cell, not loading a narrative that already contains one
+         * @param {object} typesToFiles keys = data types, values = arrays of files to import
```

Comment:
small doc note: would be helpful to have an example here of a data type, and more descriptors on the files (e.g. string which references the file in staging)

Reply:
Updated with an example.
Also, this is just a straw-person. I figure we'll need a way to bring in the object types and files that we're having the bulk importer work on, and this seemed like a first step. I'm sure there's a better way to do it.